src/main/java/org/apache/sling/commons/threads/impl/ThreadPoolMetricsGauges.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.sling.commons.threads.impl;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.management.DynamicMBean;
+import javax.management.JMException;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanInfo;
+
+import org.apache.sling.commons.metrics.Gauge;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class ThreadPoolMetricsGauges {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ThreadPoolMetricsGauges.class);
+
+    @SuppressWarnings("unchecked")
+    private static final Map<Class<?>, Class<?>> wrapperToPrimitive = new HashMap() {
+        {
+            put(Integer.class, int.class);
+            put(Long.class, long.class);
+            put(Boolean.class, boolean.class);
+        }
+    };
+
+    @SuppressWarnings("unchecked")
+    private static final Map<Class<?>, Object> clazzToNvl = new HashMap() {
+        {
+            put(Integer.class, -1);
+            put(Long.class, -1);
+            put(Boolean.class, false);
+            put(String.class, "");
+        }
+    };
+
+    private static boolean isTypeOfClassOrPrimitive(MBeanAttributeInfo attr, Class<?> clazz) {
+        if (clazz.toString().equals(attr.getType())) {
+            return true;
+        }
+
+        Class<?> primitive = wrapperToPrimitive.get(clazz);
+        if (primitive != null && primitive.toString().equals(attr.getType())) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private static <T> Gauge<T> createGauge(final DynamicMBean bean, final String name, Class<T> clazz,
+            final Object nvl) {
+        return new Gauge<T>() {
+            @SuppressWarnings("unchecked")
+            public T getValue() {
+                try {
+                    return (T) bean.getAttribute(name);
+                } catch (JMException e) {
+                    LOGGER.warn("cannot obtain MBean attribute named " + name, e);
+                    return (T) nvl;
+                }
+            }
+        };
+    }
+
+    public static Map<String, Gauge<?>> create(final DynamicMBean bean, MBeanInfo info, Collection<String> ignored) {
+        Map<String, Gauge<?>> gauges = new HashMap<>();
+        for (MBeanAttributeInfo attr : info.getAttributes()) {
+            String name = attr.getName();
+            if (ignored.contains(name)) {
+                LOGGER.debug("ignoring MBean attribute {}", name);
+            } else {
+                for (Class<?> clazz : clazzToNvl.keySet()) {
+                    if (isTypeOfClassOrPrimitive(attr, clazz)) {
```

Comment:
Can we remove this and instead just check the result of `classToNvl.get`?

Reply:
It's just an iteration over supported attributes - here key set is used indirectly instead of another explicit collection which would be exactly a copy of that key set.

Reply:
@rombert Made it more explicit with extraction of supportedClasses local set in https://github.com/apache/sling-org-apache-sling-commons-threads/pull/1/commits/490caee80eead8adea344eddedb0c3b1b080beba